mocks/auth_service_mock.go

```diff
@@ -0,0 +1,24 @@
+package mocks
+
+import (
+	"context"
+	"net/http"
+)
+
+type AuthService struct {
+	Mocker
+}
+
+func (a AuthService) GetConfiguredHttpClient(ctx context.Context) (*http.Client, error) {
+	returns := a.called("GetConfiguredHttpClient", ctx)
+	return returns.Get(0).(*http.Client), returns.Error(1)
```

Comment:
When returning errors it's necessary to use the specialized `Error` method, which takes care of returning a `error(nil)` because it's impossible to cast `nil` to an `error` with `nil.(error)`.

Other types might require this later, but I don't need them right now so 🤷🏼